### ISSUE TITLE
`buildLuarocksPackage`: `build_dependencies` and `test_dependencies` not available to luarocks during `checkPhase`

### ISSUE THREAD

**Issue by mrcjkb** [NixOS/nixpkgs#358233]
## Describe the bug

When building a luarocks package with a custom build backend (e.g. luarocks-build-rust-mlua), or that has test dependencies (e.g. `nlua`), luarocks does not pick up these dependencies from the `LUAROCKS_CONFIG` in the checkPhase.

For build backends like `luarocks-build-rust-mlua`, it is necessary to run `luarocks make` in the `preCheck` hook. This fails because luarocks tries to download the build backend from luarocks org.

The only workaround I've found so far is to add the test dependencies and build backends to the `propagatedBuildInputs`, but this means that build/test dependencies will unnecessarily be propagated.

## Steps To Reproduce

See for example [this overlay](https://github.com/Saghen/blink.cmp/pull/356/files#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0R73).

## Expected behavior

luarocks should pick up `test_dependencies` and `build_dependencies` from the `nativeCheckInputs` or `nativeBuildInputs`.

## Notify maintainers

cc @teto 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by teto** [comment]
agreed. In fact I see some reverted attempts of doing so in my nixpkgs fork. Go ahead :)

**Comment by teto** [comment]
i think it's fixed now ?

**Comment by mrcjkb** [comment]
Yep, seems to be fixed.
